ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveFilter.java

```diff
@@ -72,6 +78,22 @@ public HiveFilter(RelOptCluster cluster, RelTraitSet traits, RelNode child, RexN
     super(cluster, TraitsUtil.getDefaultTraitSet(cluster), child, condition);
     this.correlationInfos = new CorrelationInfoSupplier(getCondition());
   }
+  
+  @Override 
+  public RelWriter explainTerms(RelWriter pw) {
+    // Remove this method after upgrading Calcite to 1.35+
```

Comment:
Is it a good idea to write project properties with `properties-maven-plugin` to `ql/src/main/resources/`, and read the calcite version from there and compare? This plugin is already in use in `itests`.

If we just define a constant CALCITE_VERSION = 1.35, it won't throw an exception when the version is higher than the constant. But if we read directly from the properties file, it'll throw an exception.

Reply:
> write project properties with properties-maven-plugin to ql/src/main/resources/, and read the calcite version from there

It can work. 
However we already have constants needs to be manually modified at a Calcite upgrade:
https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/Bug.java

Reply:
For now I am adding a manual version check, however, if you want me to, I can create a new ticket to automate like I have described above.